pkg/deploy/controller/deployment/factory.go

```diff
@@ -110,6 +111,10 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 					return false
 				}
 				if retries.Count > 1 {
+					deployment, ok := obj.(*kapi.ReplicationController)
+					if ok {
+						eventRecorder.Eventf(deployment, kapi.EventTypeWarning, "FailedRetry", "About to stop retrying %s: %v", deployment.Name, err)
```

Comment:
I think we need something more fine grained.  The end goal is to only
record meaningful events - I think anything retried is not meaningful,
anything we drop out of retry should be stable (i.e, if every 2 minutes we
fire the event, the message should be the same each time) so we get
counting, and anything we *know* is a serious problem we send an event
for.  I think the only one who knows about serious errors is the
controller, and the retry logic here should be writing stable messages (the
update message is not stable, for instance, because resource version
changes each time).

On Mon, May 9, 2016 at 8:40 AM, Michail Kargakis <notifications@github.com>
wrote:

> In pkg/deploy/controller/deployment/factory.go
> <https://github.com/openshift/origin/pull/8652#discussion_r62491057>:
>
> > @@ -110,6 +111,10 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
> >  					return false
> >  				}
> >  				if retries.Count > 1 {
> > +					deployment, ok := obj.(*kapi.ReplicationController)
> > +					if ok {
> > +						eventRecorder.Eventf(deployment, kapi.EventTypeWarning, "FailedRetry", "About to stop retrying %s: %v", deployment.Name, err)
>
> Not all transient errors are about updating the deployment. Actually the
> only fatal error in the deployment controller is when the utility for
> generating a deployer pod will error out.... Should we add more error types
> and distinguish here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8652/files/963d4b3e46e3d3b21cc13791d439816dd587f46a#r62491057>
>